components/heatmap/HeatMap.vue

```diff
@@ -195,7 +214,52 @@ export default {
     ];
     this.passing_plate_colors = plate_colors;
   },
-  methods: {},
+  methods: {
+    on_update_maximum: function (new_value) {
+      const max = parseInt(new_value);
+      if (new_value == "") {
+        this.upper = 0;
+        this.max_heatmap_value = "invalid";
+      } else {
+        if (max < 0 || new_value == "-") {
+          this.max_heatmap_value = "cannot be negative";
+        } else {
+          if (this.lower > max) {
+            this.max_heatmap_value = "min greater than max";
+            this.min_heatmap_value = "min greater than max";
+          } else {
+            if (max > 1000) {
+              this.max_heatmap_value = "larger than 1000";
+            } else {
+              this.upper = max;
+              this.max_heatmap_value = "";
+              this.min_heatmap_value = "";
+            }
+          }
+        }
+      }
+    },
+    on_update_minimum: function (new_value) {
+      const min = parseInt(new_value);
+      if (new_value == "") {
+        this.lower = 0;
+        this.min_heatmap_value = "invalid";
+      } else {
+        if (min < 0 || new_value == "-") {
+          this.min_heatmap_value = "cannot be negative";
+        } else {
+          if (min > this.upper) {
+            this.max_heatmap_value = "min greater than max";
+            this.min_heatmap_value = "min greater than max";
+          } else {
```

Comment:
Yes, also specify the default starting value of min and max to start with.. then will update the data() and resolve the above.

Reply:
Making this change has absolutely nothing to do with the default values, if a user enters 100/100 it should not allow it.

Reply:
Now the value of equal Min/Max is handled, and error is reported and not set on heatmapcolorbar.

Reply:
This code still allows me to enter the same value for min/max if I enter the min value first, and if I enter min first the UI doesn't update the min value below the gradient bar.

![heatmap](https://user-images.githubusercontent.com/236104/116131276-0a603000-a681-11eb-9174-be3a43af918b.gif)

Reply:
Well, you haven't never mentioned that user has to be prevented from entry into the INPUT field, currently the UI/UX design from Ari in Mockflow also doesn't say its to be prevented from entry. 

Currently, we can control the value been applied on to the heatmapcolorbar, which you can observe in your own video grab.

So are u suggesting to prevent user from entry ?

Reply:
As per the form UI/UX principles entry is top-down approach, so User has to enter maximum value first and then enters the minimum value.

Implementing the same.

Reply:
**

https://user-images.githubusercontent.com/977583/116234998-f00a7d00-a77a-11eb-90f9-c1b5c66844d2.mov

**

Reply:
Please update the changes that I have incorporated, see the UI/UX definition is the work from Ari, I just implement what is been provided.

This project UI/UX has no wire-frame diagram and navigation flows define.

Reply:
UI/UX design mostly has navigation and interaction definition via wireframe diagrams..

Currently I am just implementing even the error response text based some reference response. There is no requirement or definition provided.

Reply:
This code still allows invalid input

![heatmap](https://user-images.githubusercontent.com/236104/116428486-cf353c80-a7f9-11eb-8d73-d9356c7c0272.gif)

This behavior is confusing, the error message doesn't get updated on the min input and it won't allow me to enter 200 unless I delete the value and retype it.

![heatmap2](https://user-images.githubusercontent.com/236104/116429117-6ef2ca80-a7fa-11eb-901b-f85b5b2a6d7b.gif)

Reply:
Currently, the UI/UX definition for all the possible sequences and errors are not specified in Mockflow UI. So Ari has to provide all the details without which I don't know what to resolve.

So as requested earlier this needs a UI/UX definition discussion. Without I don't know what to resolve.